src/state_machine/signer/mod.rs

```diff
@@ -220,6 +227,10 @@ pub struct Signer<SignerType: SignerTrait> {
     pub dkg_private_begin_msg: Option<DkgPrivateBegin>,
     /// the DKG end begin message received in this round
     pub dkg_end_begin_msg: Option<DkgEndBegin>,
+    /// Ephemeral private key for key exchange
+    kex_private_key: Scalar,
+    /// Ephemeral public keys for key exchange indexed by key_id
+    kex_public_keys: HashMap<u32, Point>,
```

Comment:
Shouldn't they be indexed by `signer_id`?

Reply:
We already have a way to go from `signer_id` to `kex_public_key`: use the `dkg_public_shares` map which is indexed on `signer_id`.  What we needed here was a way to go from `key_id` to `kex_public_key`, so we could easily encrypt the `DkgPrivateShares`.

Reply:
When we create `DkgPrivateShares`, we iterate through all of the `key_ids`, and evaluate our polynomial at that ID.  We then encrypt the polynomial evaluations for the owner (signer) of that `key_id`, and send a `Packet` which contains all of the evaluations indexed by `key_id`.  

The configuration naturally has a way to go from `signer_id` to `key_ids`, but not the reverse. So in this PR, when we receive `DkgPublicShares` from a particular `signer_id`, we look up all of the `key_ids` (using the aforementioned map), then map those `key_ids` to this round's `key_public_key` for the sending `signer_id`.  

Does that make sense?